test/metabase/test/data.clj

```diff
@@ -172,10 +172,18 @@
     :type     :native
     :native   ~inner-native-query})
 
+(def ^:dynamic *run-post-process-fn*
```

Comment:
Why don't you just name this `*post-process-fn*`, we don't "run post-process" results, we post-process results. See https://stuartsierra.com/2016/01/09/how-to-name-clojure-functions for a little more on this.

Also it's always good to add `^:arglists` metadata to plain `def`s that are functions, because then your editor can show the arguments when you're calling it and Kondo or Eastwood or whatever can check arg counts

```suggestion
(def ^:dynamic ^{:arglists '([results])} *post-process-fn*
```

Reply:
If I'm able to implement rebinding `id` function in custom `run-mbql-query` for timeseries drivers, I'd be able to completely remove this. More on that in [this comment](https://github.com/metabase/metabase/pull/40293#discussion_r1589446353). Till then, I've renamed the function as you suggested.